### PR TITLE
Add more merge processing (non substrantive)

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds more merge processing-related (non-substantive with Altair) path to the general state transition function. 

I discovered one bug where the `genesisBeaconStateMerge` test function returned the beacon state with version Altair instead of the beacon state with version Merge. I updated some test results accordingly

### CODE REVIEW

File: testing/util/merge_state.go [hunk 1]
```diff
@@ -81,3 +84,159 @@ func emptyGenesisStateMerge() (state.BeaconState, error) {
 	}
 	return v3.InitializeFromProto(st)
 }
+
+func buildGenesisBeaconStateMerge(genesisTime uint64, preState state.BeaconState, eth1Data *ethpb.Eth1Data) (state.BeaconState, error) {
```
Comment (terencechain) [inline testing/util/merge_state.go:88]: test only

File: testing/util/merge_state.go [hunk 2]
```diff
@@ -49,7 +52,7 @@ func genesisBeaconStateMerge(ctx context.Context, deposits []*ethpb.Deposit, gen
 		return nil, errors.Wrap(err, "could not process validator deposits")
 	}
 
-	return buildGenesisBeaconState(genesisTime, st, st.Eth1Data())
+	return buildGenesisBeaconStateMerge(genesisTime, st, st.Eth1Data())
```
Comment (terencechain) [inline testing/util/merge_state.go:55]: test only `genesisBeaconStateMerge` bug, `buildGenesisBeaconState` returns Altair state

File: beacon-chain/cache/sync_committee_head_state.go [hunk 3]
```diff
@@ -31,10 +32,13 @@ func (c *SyncCommitteeHeadStateCache) Put(slot types.Slot, st state.BeaconState)
 	if st == nil || st.IsNil() {
 		return ErrNilValueProvided
 	}
-	_, ok := st.(*stateAltair.BeaconState)
-	if !ok {
+
+	_, okV2State := st.(*v2.BeaconState)
+	_, okV3State := st.(*v3.BeaconState)
+	if !okV2State && !okV3State {
```
Comment (rauljordan) [inline beacon-chain/cache/sync_committee_head_state.go:38]: A more robust pattern is to just return if it's a phase 0 beacon state

File: beacon-chain/cache/sync_committee_head_state.go [hunk 4]
```diff
@@ -47,9 +51,13 @@ func (c *SyncCommitteeHeadStateCache) Get(slot types.Slot) (state.BeaconState, e
 	if !exists {
 		return nil, ErrNotFound
 	}
-	st, ok := val.(*stateAltair.BeaconState)
+	var st state.BeaconState
```
Comment (rauljordan) [inline beacon-chain/cache/sync_committee_head_state.go:54]: same thing here, check if phase 0 and return incorrect type

Comment (terencechain) [reply beacon-chain/cache/sync_committee_head_state.go:54]: For this, we need to convert the interface. Would you prefer something like the following?

```go
	_, ok := st.(*v1.BeaconState)
	if ok {
		return ErrIncorrectType
	}

	var st state.BeaconState
	st, ok := val.(*v2.BeaconState)
	if !ok {
		st, ok = val.(*v3.BeaconState)
		if !ok {
			return nil, ErrIncorrectType
		}
	}
	return st, nil
```

**Review by rauljordan — APPROVED** [review]

**Pull request merged**